pymc3/progressbar.py

```diff
@@ -117,10 +107,11 @@ def run_from_ipython():
 
 
 def progress_bar(iters):
-    if run_from_ipython():
-        if None:
-            return NotebookProgressBar(iters)
-        else:
-            return TextProgressBar(iters, ipythonprint)
-    else:
+    try:
+        from IPython.html import widgets
+        widgets.FloatProgress()
+        return IPythonNotebookPB(iters)
+    except RuntimeError:
+        return TextProgressBar(iters, ipythonprint)
+    except ImportError:
```

Comment:
Don't really like this. @jsalvatier suggested that I move this to head of the file. Still feels very hacky. 
Any thoughts?